source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2515 @@
+    will map to an array of strings, each representing a tag set, since it is
+    not feasible to define multiple ``readPreferenceTags`` keys in the document.
+
+  .. _entity_client_useMultipleMongoses:
+
+  - ``useMultipleMongoses``: Optional boolean. If true and the topology is a
+    sharded cluster, the test runner MUST assert that this MongoClient connects
+    to multiple mongos hosts (e.g. by inspecting the connection string). If
+    false and the topology is a sharded cluster, the test runner MUST ensure
+    that this MongoClient connects to only a single mongos host (e.g. by
+    modifying the connection string). This option has no effect for non-sharded
+    topologies.
+
+  .. _entity_client_observeEvents:
+
+  - ``observeEvents``: Optional string or array of strings. One or more types of
+    events that can be observed for this client. Unspecified event types MUST
+    be ignored by this client's event listeners and SHOULD NOT be included in
+    `test.expectedEvents <test_expectedEvents_>`_ for this client.
+
+    Test files SHOULD NOT observe events from multiple specs (e.g. command
+    monitoring *and* SDAM events) for a single client. See
+    `Mixing event types in observeEvents and expectedEvents`_ for more
+    information.
+
+    Supported types correspond to those documented in `expectedEvent`_ and are
+    as follows:
+
+    - `commandStartedEvent <expectedEvent_commandStartedEvent_>`_
+
+    - `commandSucceededEvent <expectedEvent_commandSucceededEvent_>`_
+
+    - `commandFailedEvent <expectedEvent_commandFailedEvent_>`_
+
+  .. _entity_client_ignoreCommandMonitoringEvents:
+
+  - ``ignoreCommandMonitoringEvents``: Optional string or array of strings. One
+    or more command names for which the test runner MUST ignore any observed
+    command monitoring events. The command(s) will be ignored in addition to
+    ``configureFailPoint`` and any commands containing sensitive information
+    (per the
+    `Command Monitoring <../command-monitoring/command-monitoring.rst#security>`__
+    spec).
+
+    Test files SHOULD NOT use this option unless one or more command monitoring
+    events are specified in `observeEvents <entity_client_observeEvents_>`_.
+
+.. _entity_database:
+
+- ``database``: Optional document. Corresponds with a Database object.
+
+  The structure of this document is as follows:
+
+  - ``id``: Required string. Unique name for this entity. The YAML file SHOULD
+    define a `node anchor`_ for this field (e.g. ``id: &database0 database0``).
+
+  - ``client``: Required string. Client entity from which this database will be
+    created. The YAML file SHOULD use an `alias node`_ for a client entity's
+    ``id`` field (e.g. ``client: *client0``).
+
+  - ``databaseName``: Required string. Database name. The YAML file SHOULD
+    define a `node anchor`_ for this field (e.g.
+    ``databaseName: &database0Name foo``).
+
+  - ``databaseOptions``: Optional document. See `collectionOrDatabaseOptions`_.
```

Comment:
In addition to my comments reg `Entity` and `Operation` consistency I think we could have a `parameters` field here that deprecates the `<prefix>Options` key, and perhaps puts `client`, and `databaseName` into parameters for something like:

```json
{
  name: `Database`,
  id: 'db-ref'
  parameters: {
    databaseName: 'test',
    clientAlias: 'client-ref',
    readPreference: {
      mode: 'secondary'
    }
  }
}
```

Reply:
The spec states `operation` has a `name` and `parameters`.

This creates an incompatibility if we desired to type the arguments / options /
parameters to a given method, which could be useful in the future. We currently
type `Client`, `Db`, `Collection` options / parameters / arguments, why not
allow a similar strict typing for methods like `findOne` and `insertOne` in the future? The
issue being that if an argument was added to one of these methods we can't issue
a spec change because the parameters for individual methods are not well
defined.

My recommendation for this is to create a shared definition format for any
executable code, some way to tie in the name of a method / class / function and
it's minimum required arguments.

```js
{
  reference: 'client',
  parameters: {
    uriOptions: {
      replicaSet: string
    }
  }
}
```

```js
{
  reference: 'findOne',
  parameters: {
    filter: { x: 42 }
  }
}
```

Reply:
I'm down to resolve this idea.

When I proposed it my philosophy is more configuration over convention, and would involve heaver driver-side-mapping, rather then directly executing.